src/AppBundle/Controller/Report/ReportController.php

```diff
@@ -25,24 +29,20 @@ public function addAction(Request $request)
 
         $reportData = $this->deserializeBodyContent($request);
 
-        // new report
         if (empty($reportData['client']['id'])) {
             throw new \InvalidArgumentException('Missing client.id');
         }
         $client = $this->findEntityBy(EntityDir\Client::class, $reportData['client']['id']);
         $this->denyAccessIfClientDoesNotBelongToUser($client);
 
-        $report = new Report($client);
-        $this->get('opg_digideps.report_service')->setReportTypeBasedOnCasrec($report);
-
         $this->validateArray($reportData, [
             'start_date' => 'notEmpty',
             'end_date' => 'notEmpty',
         ]);
 
-        // add other stuff
-        $report->setStartDate(new \DateTime($reportData['start_date']));
-        $report->setEndDate(new \DateTime($reportData['end_date']));
+        // report type is taken from CASREC. In case that's not available (shouldn't happen unless casrec table is dropped), use a 102
+        $reportType = $this->get('opg_digideps.report_service')->getReportTypeBasedOnCasrec($client) ?: Report::TYPE_102;
```

Comment:
Problem is that that method is also used for creating new year reports (PA included), where the default value (happening for PA that don't have a CASREC entry at the submit time) is the old year report's type. Report type will at some point require that casrec entry, and at that time we can refactor this without using a default value